src/supernova/mod.rs

```diff
@@ -176,92 +176,182 @@ where
   }
 }
 
-/// SuperNova takes Ui a list of running instances.
-/// One instance of Ui is a struct called RunningClaim.
-pub struct RunningClaim<G1, G2, C1, C2>
+/// A vector of [PublicParams] corresponding to a set of [RunningClaims]
+pub struct RunningClaimParams<G1, G2, C1>
 where
   G1: Group<Base = <G2 as Group>::Scalar>,
   G2: Group<Base = <G1 as Group>::Scalar>,
   C1: EnforcingStepCircuit<G1::Scalar>,
-  C2: EnforcingStepCircuit<G2::Scalar>,
 {
-  _phantom: PhantomData<C1>,
-  augmented_circuit_index: usize,
-  c_secondary: C2,
-  params: PublicParams<G1, G2>,
-}
-
-/// Indexed list of `RunningClaim`s representing an NIVC computation in process.
-pub struct RunningClaims<G1, G2, C1, C2>
-where
-  G1: Group<Base = <G2 as Group>::Scalar>,
-  G2: Group<Base = <G1 as Group>::Scalar>,
-  C1: StepCircuit<G1::Scalar>,
-  C2: StepCircuit<G2::Scalar>,
-{
-  /// Indexed `RunningClaim`s
-  claims: Vec<RunningClaim<G1, G2, C1, C2>>,
+  /// The internal public params
+  pub pp_vec: Vec<PublicParams<G1, G2>>,
   /// Digest constructed from these `RunningClaim`s' parameters
   // Once we serialize this struct, it's important to add this
   // annotaiton to the digest field for it to be treated properly: #[serde(skip, default = "OnceCell::new")]
   digest: OnceCell<G1::Scalar>,
+  stage: Stage,
+  _p: PhantomData<C1>,
```

Comment:
Why is there no parametrization by C2?

Reply:
Fixed. This was originally because `NonUniformCircuit` wasn't parametriation by `C2`, either.